manifests/profile/master/postgres_access.pp

```diff
@@ -60,13 +60,24 @@
       $postgres_version = '9.4'
     }
 
-    puppet_enterprise::pg::cert_whitelist_entry { 'allow-telegraf-access':
-      user                          => 'telegraf',
-      database                      => 'pe-puppetdb',
-      allowed_client_certname       => $_telegraf_host,
-      pg_ident_conf_path            => "/opt/puppetlabs/server/data/postgresql/${postgres_version}/data/pg_ident.conf",
-      ip_mask_allow_all_users_ssl   => '0.0.0.0/0',
-      ipv6_mask_allow_all_users_ssl => '::/0',
+    if defined(puppet_enterprise::pg::cert_allowlist_entry) {
+      puppet_enterprise::pg::cert_allowlist_entry { 'allow-telegraf-access':
+        user                          => 'telegraf',
+        database                      => 'pe-puppetdb',
+        allowed_client_certname       => $_telegraf_host,
+        pg_ident_conf_path            => "/opt/puppetlabs/server/data/postgresql/${postgres_version}/data/pg_ident.conf",
+        ip_mask_allow_all_users_ssl   => '0.0.0.0/0',
+        ipv6_mask_allow_all_users_ssl => '::/0',
+      }
+    } else {
+      puppet_enterprise::pg::cert_whitelist_entry { 'allow-telegraf-access':
+        user                          => 'telegraf',
+        database                      => 'pe-puppetdb',
+        allowed_client_certname       => $_telegraf_host,
+        pg_ident_conf_path            => "/opt/puppetlabs/server/data/postgresql/${postgres_version}/data/pg_ident.conf",
+        ip_mask_allow_all_users_ssl   => '0.0.0.0/0',
+        ipv6_mask_allow_all_users_ssl => '::/0',
+      }
```

Comment:
First off, I have no objection to what you have here. I just wanted to offer up a deduplicated version as an alternative:

```suggestion
      $telegraf_allow_settings = {
        'user'                          => 'telegraf',
        'database'                      => 'pe-puppetdb',
        'allowed_client_certname'       => $_telegraf_host,
        'pg_ident_conf_path'            => "/opt/puppetlabs/server/data/postgresql/${postgres_version}/data/pg_ident.conf",
        'ip_mask_allow_all_users_ssl'   => '0.0.0.0/0',
        'ipv6_mask_allow_all_users_ssl' => '::/0',
      }
      puppet_enterprise::pg::cert_allowlist_entry { 'allow-telegraf-access':
        * => $telegraf_allow_settings,
      }
    } else {
      puppet_enterprise::pg::cert_whitelist_entry { 'allow-telegraf-access':
        * => $telegraf_allow_settings,
      }
```

https://puppet.com/docs/puppet/6.19/style_guide.html#attribute-ordering